R/compile_report.R

```diff
@@ -231,15 +237,47 @@ compile_report <- function(file, quiet = FALSE, factory = getwd(),
   }
 
   dir.create(output_dir, FALSE, TRUE)
-
-  for (file in new_files) {
+  
+  output_files <- vector(length(new_files), mode = "list")
+  for (i in seq_along(new_files)) {
+    file <- new_files[i]
     destination <- file.path(output_dir, file)
     move_file(file, destination)
+    output_files[[i]] <- destination
   }
-
+  
   ## remove empty directories
   to_remove <- file.path(new_dirs)
   file.remove(to_remove)
+  
+  
+  # ================
+  ### ! KEEP AT END OF FUNCTION ! ###
+  ## Logging approach: 
+  ## 1. Combine all env variables and args into a list, and add timestamp 
+  ## 2. Read exising log rds file (a list of lists)
+  ## 3. Add new entry to dataframe with env() on compile_report call
+        ## output filenames, and other relevant data
+  
+  ## Start log code
+  log_file_path <- paste(factory, ".compile_log.rds", sep = "/")
+  no_log_file <- sum(file.exists(log_file_path, hidden.files = TRUE)) == 0
+  if (no_log_file) {
+    initialize_log <- list(initialize = TRUE, timestamp = Sys.time())
+    saveRDS(initialize_log, log_file_path)
+  }
+  report_source_file <- c("name" <- output_file)
+  log_entry$report_source_file <- report_source_file
+  log_entry$output_files <- output_files 
+  current_log <- readRDS(log_file_path)
+  if (is.null(current_log[[base_name]])) current_log[[base_name]]
+  
+  string_time <- format(Sys.time(), "%Y-%m-%d %H:%M:%S")
```

Comment:
That format used to name a list will be a bit problematic because of the spaces. Maybe use the format we use to name compiled outputs `"%Y-%m-%d_%H-%M-%S"`?